src/components/AddressBadge.tsx

```diff
@@ -16,54 +16,74 @@ You should have received a copy of the GNU Lesser General Public License
 along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
-import { StyleProp, TextStyle, View, ViewStyle } from 'react-native'
+import { StyleProp, TextStyle, TouchableNativeFeedback, View, ViewStyle } from 'react-native'
+import { useTheme } from 'styled-components'
 import styled from 'styled-components/native'
 
 import AppText from '~/components/AppText'
+import Button from '~/components/buttons/Button'
 import { useAppSelector } from '~/hooks/redux'
 import DefaultAddressBadge from '~/images/DefaultAddressBadge'
 import { selectAddressByHash } from '~/store/addressesSlice'
 import { AddressHash } from '~/types/addresses'
+import { copyAddressToClipboard } from '~/utils/addresses'
 
 interface AddressBadgeProps {
   addressHash: AddressHash
   hideSymbol?: boolean
   textStyle?: StyleProp<TextStyle>
+  color?: string
+  showCopyBtn?: boolean
   style?: StyleProp<ViewStyle>
 }
 
-const AddressBadge = ({ addressHash, hideSymbol = false, textStyle, style }: AddressBadgeProps) => {
+const AddressBadge = ({ addressHash, hideSymbol = false, textStyle, color, showCopyBtn, style }: AddressBadgeProps) => {
+  const theme = useTheme()
   const address = useAppSelector((s) => selectAddressByHash(s, addressHash))
 
+  const textColor = color || theme.font.primary
+
   return (
-    <View style={style}>
-      {!address ? (
-        <Label numberOfLines={1} ellipsizeMode="middle" style={textStyle}>
-          {addressHash}
-        </Label>
-      ) : (
-        <>
-          {!hideSymbol && (
-            <Symbol>
-              {address.settings.isDefault ? (
-                <DefaultAddressBadge size={16} color={address.settings.color} />
-              ) : (
-                <Dot color={address.settings.color} />
-              )}
-            </Symbol>
-          )}
-          {address.settings.label ? (
-            <Label numberOfLines={1} style={textStyle}>
-              {address.settings.label}
-            </Label>
-          ) : (
-            <Label numberOfLines={1} ellipsizeMode="middle" style={textStyle}>
-              {address.hash}
-            </Label>
-          )}
-        </>
-      )}
-    </View>
+    <TouchableNativeFeedback onLongPress={() => !showCopyBtn && copyAddressToClipboard(addressHash)}>
```

Comment:
I've passed down the `onPress` props from the `AddressBox` to the `AssetBadge`... but I'm not sure I understand why this is needed. Pressing any children without onPress prop in the AddressBox (which is the parent) should trigger the onPress callback I think...

Reply:
I think that the `longPress` does lot let the `onPress` get triggered otherwise...

> Clicking on an address hash in the Address list screen (to select an address as a destination address in the Send workflow) does nothing. I suspect this was introduced with the “long press feature”.

Can you reproduce this issue?

Reply:
Yes I could, and passing down the onPress event fixed the issue. You probably right, supporting the longPress on a children probably prevent the triggering of a "press" on the parent.